File: UI/ecinema_desktop/lib/models/actor.dart
import 'package:json_annotation/json_annotation.dart';

part 'actor.g.dart';

@JsonSerializable()
class Actor {
  int? id;
  String? firstName;
  String? lastName;
  DateTime? dateOfBirth;
  final String? biography;
  String? image;
  final bool isDeleted;

  Actor({
    this.id,
    this.firstName,
    this.lastName,
    this.dateOfBirth,
    this.biography,
    this.image,
    required this.isDeleted,
  });

  factory Actor.fromJson(Map<String, dynamic> json) => _$ActorFromJson(json);
  Map<String, dynamic> toJson() => _$ActorToJson(this);
} 

File: eCinema/eCinema.Services/Database/Entities/Actor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace eCinema.Services.Database.Entities
{
    public class Actor
    {
        public int Id { get; set; }
        
        [Required]
        [MaxLength(50)]
        public string FirstName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(50)]
        public string LastName { get; set; } = string.Empty;
        
        public DateTime? DateOfBirth { get; set; }
        
        [MaxLength(500)]
        public string? Biography { get; set; }
        
        public byte[]? Image { get; set; }

        public bool IsDeleted { get; set; } = false;
        
        public virtual ICollection<MovieActor> MovieActors { get; set; } = new HashSet<MovieActor>();
    }
} 